api/api/views/oauth2_views.py

```diff
@@ -154,7 +155,9 @@ def post(self, *args, **kwargs):
         endpoint.
         """
 
-        return super().post(*args, **kwargs)
+        res = super().post(request._request)
+        data = json.loads(res.content)
+        return Response(data, status=res.status_code)
```

Comment:
Is it necessary to `json.loads` the content? Would something like this work without the need for that?

```suggestion
        data = json.loads(res.content)
        return HttpResponse(content=res.content, content_type="application/json", status=res.status_code)
```

`HttpResponse` is the more low-level version from `django.http` that allows you to bypass the DRF `Response` data parsing. Doing `json.loads` and then passing it directly to `Response` means we're unnecessary marshalling the JSON string to Python and then immediately back to a string.

Reply:
Although, more general question: why is it necessary to do anything manual here at all, other than just return `res` like we used to? Does `APIView` complain about not receiving the right type of response object or something?

Reply:
> Using the `Response` class simply provides a nicer interface for returning content-negotiated Web API responses, that can be rendered to multiple formats.
> &mdash; DRF docs

We can use `HttpResponse` for sure, but sending DRF's Response is more consistent as it renders the DRF API UI in the browser.

Reply:
> We can use HttpResponse for sure, but sending DRF's Response is more consistent and it renders the DRF API UI in the browser like our other endpoints.

Ahh, interesting, I didn't realise that was a side effect. For this view, it doesn't seem like a tragic loss, though? It's not exactly a "browseable" response :thinking: At least not any less so if your browser just renders a nice JSON explorer (Firefox does this, not sure of others).

I also don't think we _want_ to support anyone using this view over anything than JSON anyway, but I guess `Response` handles it regardless so :shrug: I would personally use HttpResponse but if you prefer to keep it, all good with me as well :+1: